src/main/java/com/jcabi/http/wire/AbstractHeaderBasedCachingWire.java

```diff
@@ -173,13 +171,15 @@ private Response validateCacheWithServer(
     }
 
     /**
-     * Add response to cache.
+     * Add, update or evict response in cache.
      * @param req The request to be used as key
-     * @param rsp The response to add
+     * @param rsp The response to add/update
      */
     private void updateCache(final Request req, final Response rsp) {
         if (rsp.headers().containsKey(this.scvh)) {
             this.cache.put(req, rsp);
+        } else {
```

Comment:
@westonized the task says: "Evict cache entry if If-Modified-Since request responded with HTTP_OK code and no Last-Modified header". This would suggest that we should have if with these two conditions somewhere... Where it is?